src/lints/enum_no_repr_variant_discriminant_changed.ron

```diff
@@ -0,0 +1,78 @@
+SemverQuery(
+    id: "enum_no_repr_variant_discriminant_changed",
+    human_readable_name: "enum variant had its discriminant change value",
+    description: "A public enum's variant had its discriminant changed from its previous value.",
+    reference: Some("A public enum's variant had its discriminant changed from its previous value, which breaks implementors relying on the discriminant value."),
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/items/enumerations.html#assigning-discriminant-values"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        enum_name: name @output @tag
+
+                        attribute @optional {
+                            content {
+                                base @filter(op: "!=", value: ["$repr"])
+                            }
+                        }
```

Comment:
I don't think we want `@optional` here. Try adding an enum that has two non-repr attributes on it (e.g. `#[non_exhaustive]` and `#[allow(dead_code)]`) to see why.

There's a subtle difference in Trustfall between "if it has an attribute, it isn't repr" (AKA `@optional` and `@filter` like here) and "it doesn't have repr" (AKA `@fold @transform(op: "count") ...`). The difference is around what happens on edges with cardinality greater than 1: in the former case we have a "left outer join" in relational terms, and in the latter (because of the `@fold` aggregation), we don't.